File: src/main/java/com/lsd/diagram/ValidComponentName.java
package com.lsd.diagram;

import org.apache.commons.text.WordUtils;

import static java.util.Arrays.stream;
import static java.util.stream.Collectors.joining;

public class ValidComponentName {
    public static final String ILLEGAL_CHARS = "[()/]";

    public static String of(String name) {
        return stream(name
                .replaceAll(ILLEGAL_CHARS, " ")
                .split(" ")
        ).map(WordUtils::capitalize)
                .collect(joining())
                .replaceAll(" ", "");
    }
}

File: src/main/kotlin/com/lsd/diagram/ValidComponentName.kt
package com.lsd.diagram

import org.apache.commons.text.WordUtils
import java.util.*
import java.util.stream.Collectors.joining

object ValidComponentName {

    private val illegalChars = "[()/]".toRegex()

    @JvmStatic
    fun of(name: String): String {
        return Arrays.stream(
            name
                .replace(illegalChars, " ")
                .split(" ")
                .dropLastWhile { it.isEmpty() }.toTypedArray()
        ).map { str: String? -> WordUtils.capitalize(str) }
            .collect(joining())
            .replace(" ", "")
    }
}